research/conditional/utils/model_utils.py

```diff
@@ -365,6 +365,7 @@ def retrieve_additional_losses(model: torch.nn.Module):
         load_balancing_losses = torch.stack(load_balancing_losses)
         load_balancing_loss = torch.mean(load_balancing_losses)
         losses["load_balancing_loss"] = load_balancing_loss
+        model.forward_pass_cache["load_balancing_losses"] = []
```

Comment:
we clear the fw pass cache after every step. What is the purpose of this?

Reply:
ah, I see - gradient accumulation. It has to be empty for the tensors connected to aux loss to be freed after the optimize step